src/Psl/Type/cast.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Type;
+
+use Psl\Type\Exception\AssertException;
+use Psl\Type\Exception\CoercionException;
+
+/**
+ * @template T
+ *
+ * @param mixed $data
+ * @param TypeInterface<T> $type
+ *
+ * @return T
+ * @throws CoercionException
+ */
+function cast($data, TypeInterface $type)
+{
+    try {
+        return $type->assert($data);
+    } catch (AssertException $e) {
+    }
+
+    return $type->coerce($data);
+}
```

Comment:
hm, actually, let's not add this, and use `$type->coerce` in both regex and json. all `coerce` implementations will return the same result as assert, so i don't think we really need to assert-then-coerce.